src/coreclr/src/jit/rangecheck.cpp

```diff
@@ -444,7 +444,12 @@ LclSsaVarDsc* RangeCheck::GetSsaDefAsg(GenTreeLclVarCommon* lclUse)
         return nullptr;
     }
 
-    LclSsaVarDsc* ssaDef = m_pCompiler->lvaGetDesc(lclUse)->GetPerSsaData(ssaNum);
+    LclVarDsc* varDsc = m_pCompiler->lvaGetDesc(lclUse);
+    if (varDsc->CanBeReplacedWithItsField(m_pCompiler))
+    {
+        varDsc = m_pCompiler->lvaGetDesc(varDsc->lvFieldLclStart);
+    }
+    LclSsaVarDsc* ssaDef = varDsc->GetPerSsaData(ssaNum);
```

Comment:
Perhaps you could factor this out, e.g. into `lvaGetDescForSSA` that takes a `GT_LCL_VAR` or a lclNum.

Reply:
I was thinking about it and made a prototype, but would prefer to postpone it if we could to the next PR and have a more precise design discussion there.